packages/react-router-dom/index.tsx

```diff
@@ -409,7 +409,9 @@ export const Link = React.forwardRef<HTMLAnchorElement, LinkProps>(
     },
     ref
   ) {
-    let href = useHref(to, { relative });
+    let toString = typeof to === "string" ? to : createPath(to);
+    let isExternal = toString.startsWith("//") || /^[a-z]+:/.test(toString);
```

Comment:
Should we use the same regex from https://github.com/remix-run/react-router/pull/9829?  I _think_ the same check would work for both of these scenarios